controllers/humiocluster_controller.go

```diff
@@ -2250,6 +2314,13 @@ func (r *HumioClusterReconciler) verifyHumioClusterConfigurationIsValid(ctx cont
 }
 
 func (r *HumioClusterReconciler) cleanupUnusedResources(ctx context.Context, hc *humiov1alpha1.HumioCluster, humioNodePools HumioNodePoolList) (reconcile.Result, error) {
+	if !hc.DeletionTimestamp.IsZero() {
+		if err := r.handlePDBFinalizers(ctx, hc); err != nil {
+			return r.updateStatus(ctx, r.Client.Status(), hc, statusOptions().
+				withMessage(fmt.Sprintf("failed to handle PDB finalizers: %s", err)))
+		}
+	}
```

Comment:
I believe `handlePDBFinalizers` can be removed given that all it does is remove the finalizer. 

As long as `ownerReferences` on the PDB object is configured to point at the `HumioCluster` object, then the PDB's are garbage collected once the `HumioCluster` object is deleted, see https://kubernetes.io/docs/concepts/architecture/garbage-collection/#:~:text=Cascading%20deletion,a%20process%20called%20cascading%20deletion.

Reply:
Thanks for correcting me. Conditional statement removed.